test/integration/log_system.cc

```diff
@@ -774,6 +774,18 @@ TEST_F(LogSystemTest, GZ_UTILS_TEST_DISABLED_ON_WIN32(RecordAndPlayback))
   // Start server
   Server playServer(playServerConfig);
 
+  // Simulate a client
+  gz::transport::Node node;
+  std::size_t numMsgs = 0;
```

Comment:
```suggestion
  std::atomic<std::size_t> numMsgs = 0;
```

Since a callback is invoked from a separate thread.

Reply:
[c603ad6](https://github.com/gazebosim/gz-sim/pull/2619/commits/c603ad65a0b37f8a235ac37c56af7af435596fd3)